packages/vite-plugin-svelte/src/utils/options.js

```diff
@@ -199,13 +199,18 @@ export function resolveOptions(preResolveOptions, viteConfig, cache) {
 			dev: !viteConfig.isProduction
 		}
 	};
+	const hot =
+		!viteConfig.isProduction &&
+		!preResolveOptions.isBuild &&
+		viteConfig.server &&
```

Comment:
Something I just noticed but not a big deal to change. It looks like `viteConfig.server` is always truthy, but this line implies otherwise, which could be confusing because we don't want to disable HMR if the server is not configured (HMR would've been enabled by default in dev)